go-controller/pkg/ovn/acl/acl.go

```diff
@@ -71,83 +32,41 @@ func RemoveACLFromNodeSwitches(switches []string, aclUUID string) error {
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "port_group", clusterPortGroupUUID, "acls", "@reject-acl"}
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster port group %s, stderr: %q", aclUUID, aclName, clusterPortGroupUUID, stderr)
-	}
-	return aclUUID, nil
-}
-
-// AddRejectACLToLogicalSwitch adds a reject ACL to a logical switch
-func AddRejectACLToLogicalSwitch(logicalSwitch, aclName, sourceIP string, sourcePort int, proto v1.Protocol) (string, error) {
-	l3Prefix := "ip4"
-	if utilnet.IsIPv6String(sourceIP) {
-		l3Prefix = "ip6"
-	}
-
-	aclMatch := fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
-		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=" + types.DirectionFromLPort, "priority=" + types.DefaultDenyPriority, aclMatch, "action=reject",
-		fmt.Sprintf("name=%s", aclName), "--", "add", "logical_switch", logicalSwitch, "acls", "@reject-acl"}
-
-	aclUUID, stderr, err := util.RunOVNNbctl(cmd...)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to add ACL: %s, %q, to cluster switch %s, stderr: %q", aclUUID, aclName, logicalSwitch, stderr)
+func PurgeRejectRules(nbClient libovsdbclient.Client) error {
+	// TODO
+	// - Removing blindly reject ACLs might be problematic if in the future we add this types of ACLs for other purposes
+	// - Do ACLs need to be removed or only references to them, which is what we seem to be doing elsewhere?
+	acls := []nbdb.ACL{}
+	err := nbClient.WhereCache(func(acl *nbdb.ACL) bool {
+		return acl.Action == nbdb.ACLActionReject
+	}).List(&acls)
+	if err != nil && err != libovsdbclient.ErrNotFound {
+		return errors.Wrapf(err, "Error while querying ACLs with reject action: %v", err)
 	}
-	return aclUUID, nil
-}
 
-func PurgeRejectRules(clusterPortGroupUUID string) error {
-	data, stderr, err := util.RunOVNNbctl("--columns=_uuid", "--format=csv", "--data=bare", "--no-headings", "find", "acl", "action=reject")
-	if err != nil {
-		return errors.Wrapf(err, "Error while querying ACLs with reject action: %s", stderr)
-	}
-	if strings.TrimSpace(data) == "" {
+	if len(acls) == 0 {
 		klog.Info("No reject ACLs to remove")
 		return nil
 	}
 
-	for _, uuid := range strings.Split(data, "\n") {
-		err = RemoveACLFromPortGroup(uuid, clusterPortGroupUUID)
+	aclPtrs := []*nbdb.ACL{}
+	for _, acl := range acls {
+		aclPtrs = append(aclPtrs, &acl)
+		data, stderr, err := util.RunOVNNbctl("--format=csv", "--data=bare", "--no-headings", "--columns=_uuid", "find", "logical_switch", fmt.Sprintf("acls{>=}%s", acl.UUID))
```

Comment:
This has some unit tests dependency for logical switch ops being migrated to libovsdb so I decided to wait until that work is done for this specific code. I have this in another commit for another PR and it does work at runtime and e2e tests, so it's just some unit test issue where the logical switch needs to be created as well with libovsdb instead of fake exec or go-ovn mocks.